demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -625,6 +655,87 @@ static void handleIncomingPublish( MQTTPublishInfo_t * pPublishInfo,
 
 /*-----------------------------------------------------------*/
 
+static void updateSubscribeStatus( MQTTPacketInfo_t * pPacketInfo )
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    uint8_t subscriptionStatus = 0;
+
+    assert( pPacketInfo != NULL );
+    assert( pPacketInfo->pRemainingData != NULL );
+
+    uint8_t * pPayload = NULL;
+    size_t pSize = 0;
+
+    mqttStatus = MQTT_GetSubAckStatusCodes( pPacketInfo, &pPayload, &pSize );
+
+    /* Demo only subscribes to one topic, so only one status code is returned. */
+    subscriptionStatus = pPayload[ 0 ];
+    globalSubscribeStatus = subscriptionStatus;
```

Comment:
`subscriptionStatus` isn't necessary, all variables should be declared before any asserts, and it looks like the build is failing since `mqttStatus` is never used. Actually, since `MQTT_GetSubAckStatusCodes` can't segfault I don't think the asserts are all that necessary. This function can save a few lines:
```suggestion
    uint8_t * pPayload = NULL;
    size_t pSize = 0;
    MQTTStatus_t mqttStatus = MQTT_GetSubAckStatusCodes( pPacketInfo, &pPayload, &pSize );
    
    /* The above function always returns success if called with a packet info from the event
     * callback and non-NULL parameters. */
    assert( mqttStatus == MQTTSuccess );

    /* Demo only subscribes to one topic, so only one status code is returned. */
    globalSubscribeStatus = pPayload[ 0 ];
```
I'm unsure if `mqttStatus` still needs to be `void`ed when asserts are disabled. Might as well add it too to be safe

Reply:
Fixed! Thanks for clarifying, was unsure about certain style guidelines.